storage/allocator.go

```diff
@@ -244,6 +244,57 @@ func (a *allocator) RebalanceTarget(required proto.Attributes, existing []proto.
 	return s
 }
 
+// RemoveTarget returns a suitable replica to remove from the provided replica
+// set. It attempts to consider which of the provided replicas would be the best
+// candidate for removal.
+func (a *allocator) RemoveTarget(existing []proto.Replica) (proto.Replica, error) {
```

Comment:
Added the TODO.

proto.Replica is a small struct (three ints), I think passing by value is a good choice.  

In general, I don't think we have established firm guidance in passing pointers vs. values; returning a pointer here would match the other signatures of this particular file, but it would not match the pattern used for proto.Replica elsewhere (it is typically passed by value).